osv/purl_helpers.py

```diff
@@ -63,7 +69,82 @@ def package_to_purl(ecosystem: str, package_name: str) -> str | None:
   return f'pkg:{purl_type}/{_url_encode(package_name)}{suffix}'
 
 
-def purl_to_ecosystem(purl_type: str) -> str | None:
-  """Convert purl to a specific ecosystem string"""
-  ecosystem_purl = {v: k for k, v in PURL_ECOSYSTEMS.items()}
-  return ecosystem_purl.get(purl_type)
+def parse_purl(purl_str: str) -> Purl | None:
+  """Parses a PURL string and extracts
+  ecosystem, package, and version information.
+
+  Args:
+    purl_str: The Package URL string to parse.
+
+  Returns:
+    A tuple containing the ecosystem, package,
+    and version, or None if parsing fails.
+  """
+
+  purl = PackageURL.from_string(
+      purl_str)  # May raise ValueError for invalid PURL strings
+
+  package = purl.name
+  version = purl.version
+
+  match purl:
+    case PackageURL(type='bitnami'):
```

Comment:
using match/case here is very neat! But, it does introduce 2 sources of truths for the purl type -> osv ecosystem mapping. 

would it be better to create a `PURL_TYPE_TO_OSV_ECOSYSTEM` dict automatically from  `PURL_ECOSYSTEMS` above? 

And then we can use it here just by doing a dictionary lookup on the parsed `purl`s type value?

Reply:
The existing `PURL_ECOSYSTEMS` is outdated and insufficient to support all Linux distros. My previous implementation used dictionaries, but required three separate mappings:
- PURL type to ecosystem for language ecosystems.
- PURL namespace to ecosystem for Linux distros.
- PURL namespace to type for Linux distros. (in case queries like `pkg:badtype/ubuntu/package`)
Also we have some special handling for Go, and it's best keep using a match/case statement.

I can update the `package_to_purl` function in a followup PR to try using common dictionaries or keep the same structures  with the new `parse_purl` function.

Reply:
ah, I see, thanks! if there was a way to consolidate the source of truth a bit more in another PR, that would be ideal.

Reply:
Yep, I'll do that in a separate PR. I wanted to avoid any unrelated changes in this one (in case anything breaks).